### PR TITLE
Record version info

### LINKED ISSUE #440: Record the Blades of Exile version producing the action log
This should help with debugging

### PR DESCRIPTION
**Pull request by NQNStudios**
Close #440 

I found a bug where scons builds defined GIT_TAG as empty because of a trailing blank line in the subprocess output.

Fixed that, and am now recording a root element like this:

```
<actions SHA="f411a46" Tag="v0.0.4" Status="modified" Repo="git@github.com:NQNStudios/cboe.git">
```

The CI will show whether my changes to `gitrev.sh` work for the Xcode build.

### CODE REVIEW

**Comment by NQNStudios** [comment]
Seems like Visual Studio builds don't generate tools/gitrev.hpp

**Comment by NQNStudios** [comment]
wait, there's a ClInclude for it...

**Comment by CelticMinstrel** [comment]
What's v0.0.4?

**Comment by NQNStudios** [comment]
v0.0.4 is a tag in my fork.

**Comment by NQNStudios** [comment]
Wow, I added this guard which is the same as other places gitrev.hpp is included:

```
#ifndef MSBUILD_GITREV
#include "tools/gitrev.hpp"
#endif
```

and it didn't work!

**Comment by CelticMinstrel** [comment]
Maybe the project file is configured to only define MSBUILD_GITREV in one file?

File: src/tools/replay.cpp [hunk 1]
```diff
@@ -50,10 +52,14 @@ bool init_action_log(std::string command, std::string file) {
 		log_file = file;
 		try {
 			Element root_element("actions");
+			#if defined(GIT_REVISION) && defined(GIT_TAG)
 			root_element.SetAttribute("SHA", GIT_REVISION);
 			root_element.SetAttribute("Tag", GIT_TAG);
+			#endif
+			#if defined(GIT_STATUS) && defined(GIT_REPO)
```
Comment (CelticMinstrel) [inline src/tools/replay.cpp:59]: Is there a reason for doing these in pairs?

**Comment by NQNStudios** [comment]
I think it's because I thought the Visual Studio build had a special way of defining the values that I hadn't implemented the new ones for. But now I'm pretty sure Visual Studio builds just don't have any of the values defined. I do need to extend the no-git-repo case in the SConstruct to define blank strings for the new values though, then this can be a single ifdef

**Comment by NQNStudios** [comment]
or maybe each one should have its own guard. i don't know.

**Comment by NQNStudios** [comment]
@CelticMinstrel Is there anything else you want to see from this one?

**Comment by CelticMinstrel** [comment]
Hmm… does the Visual Studio build have this information at all? Does it show the commit hash on the startup screen?

**Comment by NQNStudios** [comment]
It does not.

**Comment by CelticMinstrel** [comment]
That is _quite_ unfortunate… but I'm not sure fixing that is in scope for a PR whose purpose is just to add that info to the replay…

It's possible to code the generation of this information directly into the Visual Studio project file. I'm not sure if this approach would survive saving the project from Visual Studio, though.

**Pull request merged**